platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,246 @@
+Visual Regression Testing would help automate this process if it were in place.
+### Logging
+
+N/A
+
+### Debugging
+
+Since this project isn't about changing functionality of the forms library, we don't need to add any debugging tools. Validating that the app code can still access the forms library in its new location will be enough.
+
+### Caveats
+
+Though we will be creating a package for the forms library, we will not be following semantic versioning since it is _just_ a package and not a published module.
+The additional overhead of updating the forms-library's `package.json` file and updating the `yarn.lock` file in `vets-website` to target the newest (but unpublished) version would be a burden without much benefit.
+
+### Security Concerns
+
+N/A
+
+### Privacy Concerns
+
+N/A
+
+### Open Questions and Risks
+
+#### Do we want to run forms library unit tests on each CI build+deploy
+
+The current plan is to run forms library unit tests each time the forms library source is changed. Would we gain any additional protection by also running these on each deploy?
+
+### Work Estimates
+
+Note: These estimates do not account for extensive support requests
+
+#### Preparation
+
+**Estimate:** < 1 day
+
+1. Move the app-specific code mentioned in [Application Entanglement](#application-entanglement) outside the forms library directories.
+1. Use the `no-unresolved-modules` ESLint plugin to [find & remove unused code](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8763) in the `forms` and `forms-library` directories
+
+#### Beginning the bundling
+
+**Estimate:** ~ 4 days
+
+1. Copy what remains from `forms` into `forms-system`
+1. Create `platform/packages/forms-library` and move the contents of `platform/forms-system` into it
+1. Add a `package.json` file to the new `forms-library` package directory
+    - Any dependencies that are exclusively used by the forms library will go in here, and a `yarn install` from the `vets-website` root will install them.
+    - Shared dependencies (like `react` and `react-redux`) will be configured to be [peer dependencies](https://flaviocopes.com/npm-peer-dependencies/)
+1. [Add a temporary alias to the `.babelrc` file](https://github.com/department-of-veterans-affairs/vets-website/blob/055d96c54e1df54138b9efc589b98e55962333b3/.babelrc#L50-L55) that redirects imports from `platform/forms` and `platform/forms-system` to the new package
+1. Edit the webpack config & use a custom `sass-loader` importer to _temporarily_ rewrite imports from `forms` to `forms-system`
+    - This will prevent webpack from failing when it can't resolve the `.scss` files
+1. Fix broken relative imports
+1. Add babel configuration to the new package because according to [Babel's docs](https://babeljs.io/docs/en/config-files):
+    > Searching [for a config] will stop once a directory containing a `package.json` is found, so a relative config only applies within a single package
+
```

Comment:
We also need to add the forms library as a dependency in the root `package.json` as well, right?